src/EnergyPlus/WaterCoils.cc

```diff
@@ -5136,87 +5135,66 @@ void ReportWaterCoil(EnergyPlusData &state, int const CoilNum)
+                                                                                      waterCoil.Name,
+                                                                                      coilObjClassName,
+                                                                                      waterCoil.DesWaterCoolingCoilRate,
+                                                                                      -999.0,
+                                                                                      waterCoil.DesAirVolFlowRate,
+                                                                                      waterCoil.MaxWaterVolFlowRate);
+                waterCoil.reportCoilFinalSizes = false;
             }
         }
     }
-    ReportingConstant = state.dataHVACGlobal->TimeStepSys * DataGlobalConstants::SecInHour;
+    Real64 ReportingConstant = state.dataHVACGlobal->TimeStepSys * DataGlobalConstants::SecInHour;
     // report the WaterCoil energy from this component
-    state.dataWaterCoils->WaterCoil(CoilNum).TotWaterHeatingCoilEnergy =
-        state.dataWaterCoils->WaterCoil(CoilNum).TotWaterHeatingCoilRate * ReportingConstant;
-    state.dataWaterCoils->WaterCoil(CoilNum).TotWaterCoolingCoilEnergy =
-        state.dataWaterCoils->WaterCoil(CoilNum).TotWaterCoolingCoilRate * ReportingConstant;
-    state.dataWaterCoils->WaterCoil(CoilNum).SenWaterCoolingCoilEnergy =
-        state.dataWaterCoils->WaterCoil(CoilNum).SenWaterCoolingCoilRate * ReportingConstant;
+    waterCoil.TotWaterHeatingCoilEnergy = waterCoil.TotWaterHeatingCoilRate * ReportingConstant;
+    waterCoil.TotWaterCoolingCoilEnergy = waterCoil.TotWaterCoolingCoilRate * ReportingConstant;
+    waterCoil.SenWaterCoolingCoilEnergy = waterCoil.SenWaterCoolingCoilRate * ReportingConstant;
 
     // report the WaterCoil water collection to water storage tank (if needed)
 
-    if (state.dataWaterCoils->WaterCoil(CoilNum).CondensateCollectMode == state.dataWaterCoils->CondensateToTank) {
+    if (waterCoil.CondensateCollectMode == state.dataWaterCoils->CondensateToTank) {
         // calculate and report condensation rates  (how much water extracted from the air stream)
         // water volumetric flow of water in m3/s for water system interactions
         //  put here to catch all types of DX coils
-        Tavg = (state.dataWaterCoils->WaterCoil(CoilNum).InletAirTemp - state.dataWaterCoils->WaterCoil(CoilNum).OutletAirTemp) / 2.0;
-
-        RhoWater = GetDensityGlycol(state,
-                                    state.dataPlnt->PlantLoop(state.dataWaterCoils->WaterCoil(CoilNum).WaterPlantLoc.loopNum).FluidName,
-                                    Tavg,
-                                    state.dataPlnt->PlantLoop(state.dataWaterCoils->WaterCoil(CoilNum).WaterPlantLoc.loopNum).FluidIndex,
-                                    RoutineName);
+        Real64 Tavg = (waterCoil.InletAirTemp + waterCoil.OutletAirTemp) / 2.0;
```

Comment:
corrected average water temperature calculation for evaluating water density.

Reply:
👍